fix compile error on generic `impl` traits within types

mellowagain opened the pull request:
previously, we only checked if the outer type was an `impl Trait`. now we also check the inner type, if such one exists

fixes https://github.com/autometrics-dev/autometrics-rs/issues/121

gagbo reviewed (approved):
LGTM! Thanks for the test

emschwartz reviewed (approved):
Looks good. Can you add a changelog entry?

mellowagain commented:
> im actually running into compile errors with a local project using this very autometrics branch, so not yet merging this until i figured out that bug

the bug is now fixed. turns out i didnt correctly insert `::` between path segments. i added tests to catch such a thing in the future
